lib/universal-search.js

```diff
@@ -149,6 +156,17 @@ Search.prototype = {
     });
     app.recommendationServer.init();
 
+    // NOTE: order matters here, the urlbar and highlightManager both need
+    // to be handed the initialized recommendation
+    app.recommendation = new app.Recommendation({
+      eTLD: Services.eTLD,
+      events: app.events,
+      io: Services.io,
+      win: win,
+      RecommendationRow: app.RecommendationRow
```

Comment:
passing the constructor into the constructor, rather than fishing it out of `win.universalSearch`